projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,661 @@
+
+    /** @hidden */
+    @ViewChild(IgxToggleDirective)
+    public toggleDirective: IgxToggleDirective;
+
+    /** @hidden */
+    @ContentChildren(IgxPickerToggleComponent, { descendants: true })
+    public toggleComponents: QueryList<IgxPickerToggleComponent>;
+
+    /** @hidden */
+    @ContentChildren(IgxInputGroupBase)
+    public projectedInputs: QueryList<IgxInputGroupBase>;
+
+    @ContentChild(IgxLabelDirective)
+    public label: IgxLabelDirective;
+
+    /** @hidden */
+    @ContentChild(IgxDateSeparatorDirective, { read: TemplateRef })
+    public dateSeparatorTemplate: TemplateRef<any>;
+
+    /** @hidden */
+    public dateSeparator = CurrentResourceStrings.RangeDatePickerResStrings.igx_range_date_picker_date_separator;
+
+    /** @hidden @internal */
+    public get appliedFormat() {
+        // TODO: apply formatter if present
+        if (!this.hasProjectedInputs) {
+            // TODO: use displayFormat - see how shortDate, longDate can be defined
+            return this.inputFormat
+                ? `${this.inputFormat} - ${this.inputFormat}`
+                : `${DEFAULT_INPUT_FORMAT} - ${DEFAULT_INPUT_FORMAT}`;
+        } else {
+            return this.inputFormat ? this.inputFormat : DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public get hasProjectedInputs(): boolean {
+        return this.projectedInputs.some(i => i instanceof IgxDateStartComponent || i instanceof IgxDateEndComponent);
+    }
+
+    private _value: DateRange;
+    private $destroy = new Subject();
+    private $toggleClickNotifier = new Subject();
+    private _onChangeCallback: (_: any) => void;
+    private _positionSettings: PositionSettings;
+    private _dialogOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: true,
+        positionStrategy: new GlobalPositionStrategy()
+    };
+    private _dropDownOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: false
+    };
+
+    constructor(public element: ElementRef) {
+        this._onChangeCallback = (_: any) => { };
+    }
+
+    /**
+     * Opens the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range #dateRange mode="dialog"></igx-date-range>
+     *
+     * <button (click)="dateRange.open()">Open Dialog</button
+     * ```
+     */
+    public open(): void {
+        const range: Date[] = [];
+        if (this.value) {
+            if (this.value.start) {
+                range.push(this.value.start);
+                if (this.value.end) {
+                    range.push(this.value.end);
+                }
+            } else {
+                this.value = { start: null, end: null };
+            }
+        }
+        if (range.length > 0) {
+            this.value = { start: range[0], end: range[range.length - 1] };
+        }
+
+        this.updateCalendar();
+        if (this.mode === InteractionMode.Dialog) {
```

Comment:
Since the mode will always be either `InteractionMode.Dialog` OR `InteractionMode.DropDown`, you can refactor this whole part and drop the `openToggle` method (since it's only used here).
```
if (!this.toggleDirective.collapsed) {
    const settings = this.mode === InteractionMode.DropDown ? this._dropDownOverlaySettings : this._dialogOverlaySettings;
   this.toggleDirective.open(settings);
}
```
If you decide to add an `overlaySettings` parameter to the method, you'll only have to `Object.assign` once